text/0000-expand-math.md

```diff
@@ -0,0 +1,230 @@
+- Feature Name: Introduce new numerics -- `Rational`, `BigInt`, `BigFloat`, `Complex`
+- Start Date: 2022-02-28
+- RFC PR:
+- Pony Issue:
+
+# Summary
+
+This RFC proposes the introduction of new numeric types; in particular the addition of a type representing a fractional number (`Rational`), arbitrary precision integer (`BigInt`), arbitrary precision float (`BigFloat`), and complex number (`Complex`).
+
+# Motivation
+
+The primary motivation for adding these types to the stdlib is to have a single canonical implementation of them which allow interoperability of numeric types across the Pony ecosystem.
+
+# Detailed design
+
+I propose we add the aforementioned numeric types into `builtin` so they exist alongside the other standard numeric types. These introduced numeric types **must** existing within the current numeric type hierarchy by being compliant with existing numeric traits.
+
+## Numeric Hierarchy
+
+Current the Pony numeric type hierarchy is as follows:
+
+```mermaid
+classDiagram
+class Any
```

Comment:
At the moment, the mermaid diagrams are distracting. Is there a way to see them rendered? If not, then as part of the RFC text, I would prefer something else as a way to express whatever I am supposed to be getting from this.